Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,23 +24,48 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+
+    return ampHour_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+
+    if(current_<0){
+
+        return true;
+    }
     return false;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
-{
-    return QTime::currentTime();
+{    QTime charging;
+     charging.setHMS(0,0,0,0);
+
+    if(isCharging()==true){
+    charging= charging.addMSecs(qAbs(ampHour_/current_*3600000));
+
+     return charging;
+    }
+
+   charging = charging.addMSecs(qAbs(( BATTERY_AMP_HOUR_CAPACITY-ampHour_)/current_*3600000));
+    return charging;
 }
 
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
 {
     Q_UNUSED(batteryData);
-    // This is where you can update your variables
-    // Hint: There are many different ways that the totalAmpHoursUsed can be updated
-    // i.e: Taking a running average of your data values, using most recent data points, etc.
+
+    voltage_ = batteryData.voltage;
+    current_ = batteryData.current;
+
+    hour_= batteryData.time.hour()+batteryData.time.minute()/60.0+batteryData.time.second()/3600.0+batteryData.time.msec()/3600000.0;
```

Comment:
see above comment, also, how could you clean this up to be more reader-friendly?